### PR TITLE
PHPCS Server

### PR DESCRIPTION
**Pull request by rheinardkorf**

### CODE REVIEW

File: .env.dist [hunk 1]
```diff
@@ -1,47 +1,88 @@
...
+# S3 Storage
+LH_S3_REGION=<S3_REGION>
+LH_S3_KEY=<S3_KEY>
+LH_S3_SECRET=<S3_SECRET>
+LH_S3_BUCKET_NAME=<S3_BUCKET>
 
-# LH Server settings
+# Server settings
 LH_CONCURRENT_AUDITS=5
 LH_DEFAULT_REQUEST_CLIENT=wporg
```
Comment (valendesigns) [inline .env.dist]: The `LH_DEFAULT_REQUEST_CLIENT` variable is not being used in the binary. And there is not a `PHPCS_DEFAULT_REQUEST_CLIENT` to match the behavior in the PHPCS Server.

File: .env.dist [hunk 2]
```diff
@@ -1,47 +1,88 @@
...
 TIDE_API_AUTH_URL=http://tide.local/api/tide/v1/auth
 TIDE_API_HOST=tide.local
 TIDE_API_PROTOCOL=http
 TIDE_API_KEY=<YOUR_API_KEY>
 TIDE_API_SECRET=<YOUR_API_SECRET>
 TIDE_API_VERSION=v1
 
-# Default user in Tide API to be credited.
-# Omit to use authenticated client user.
-TIDE_API_CREDITED_USER=<USER_LOGIN_NAME>
```
Comment (valendesigns) [inline .env.dist:11]: Is this not required anymore?

File: cmd/phpcs-server/main.go [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+
+	// The client login id in Tide API.
+	flagClient = &[]string{"wporg"}[0]
+
+	// Send to Stdout rather than API?
+	flagOutput = &[]string{""}[0]
+
+	/** Processes **/
+	// A slice of processes (in order) that need to be performed on a message.
+	processes = []audit.Processor{
```
Comment (valendesigns) [inline cmd/phpcs-server/main.go:119]: Does this mean we are no longer making the server config based? I only see two types of PHPCS audits supported or does this mean that we only support post processing for these two?

File: cmd/phpcs-server/main.go [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+
+	// Initialise result with:
+	// - Tide client reference,
+	// - Temp folder for downloaded/extracted files,
+	// - Audits to run
+	result := &audit.Result{
+		"client":     &client,
+		"tempFolder": tempFolder,
+		"audits": []string{
+			"phpcs_wordpress",
```
Comment (valendesigns) [inline cmd/phpcs-server/main.go:295]: Same here, we are only allowing two types of audits. Is this not flexible anymore?

File: .env.dist (hunk 1 above)
Comment (rheinardkorf) [reply .env.dist]: Oops.... this is redundant now.  There was no way to handle this appropriately when we abstracted. The request client is in the message and if its not, then it uses the API user.  It actually does not make sense to set a default anymore.   The Sync Server ads the user in the message.

File: .env.dist (hunk 2 above)
Comment (rheinardkorf) [reply .env.dist:11]: Nope.

File: .env.dist (hunk 1 above)
Comment (valendesigns) [reply .env.dist]: I removed it

**Review by valendesigns — APPROVED** [review]

**Pull request merged**